src/main/java/io/debezium/connector/spanner/task/LowWatermarkCalculationJob.java

```diff
@@ -84,16 +85,25 @@ private Thread createMainThread() {
 
     private Thread createCalculationThread() {
         Thread thread = new Thread(() -> {
+            long lastMonitoredTime = new Date().getTime();
+            boolean printOffsets = false;
             while (true) {
 
+                long now = new Date().getTime();
+                long lag = now - lastMonitoredTime;
+                if (lag > 60 * 1000) {
```

Comment:
Extract constant as duration. All of them are 10 minutes, is it intentional? If yes maybe they should share single constant definition?
Also please replace this code with `Metronome` or `Stopwatch` class.